airbyte-integrations/bases/base-normalization/normalization/transform_config/transform.py

```diff
@@ -121,12 +121,13 @@ def transform_snowflake(self, config: dict):
         dbt_config["type"] = "snowflake"
         # here account is everything before ".snowflakecomputing.com" as it can include account, region & cloud environment information)
         dbt_config["account"] = config["host"].replace(".snowflakecomputing.com", "")
-        dbt_config["user"] = config["username"]
+        # snowflake coerces most of these values to uppercase, but if dbt has them as a different casing it has trouble finding the resources it needs. thus we coerce them to upper.
```

Comment:
this doesn't mess with other warehouses like BQ or PG?

Reply:
nope. if you expand the diff more the switch statement is by db type so this will only effect snowflake. one of the few merits of this hack :D